addons/project_sale_expense/tests/test_project_sale_expense.py

```diff
@@ -0,0 +1,39 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.hr_expense.tests.common import TestExpenseCommon
+from odoo.addons.sale.tests.common import TestSaleCommon
+from odoo.tests import Form, tagged
+
+
+@tagged('post_install', '-at_install')
+class TestSaleExpense(TestExpenseCommon, TestSaleCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+    def test_analytic_account_expense_policy(self):
+        with Form(self.product_a.product_tmpl_id) as product_form:
+            product_form.can_be_expensed = True
+            product_form.expense_policy = 'cost'
+            product_form.can_be_expensed = False
+
+            self.product_a.product_tmpl_id = product_form.save()
```

Comment:
No need to put the form in with statement if we need to get the record after saving the form. 🙂 
```suggestion
        product_form = Form(self.product_a.product_tmpl_id):
        product_form.can_be_expensed = True
        product_form.expense_policy = 'cost'
        product_form.can_be_expensed = False

        self.product_a.product_tmpl_id = product_form.save()
```